sdks/java/io/snowflake/src/test/java/org/apache/beam/sdk/io/snowflake/test/StreamingSnowflakeIOIT.java

```diff
@@ -0,0 +1,195 @@
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.testing.TestStream;
+import org.joda.time.Duration;
+import org.joda.time.Instant;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+
+/**
+ * Test for streaming in CI. Has to be run with DirectRunner. Requires a snowpipe with copy into
+ * STREAMING_IOIT table
+ *
+ * <p>Example run:
+ *
+ * <pre>
+ * ./gradlew --info -p sdks/java/io/snowflake integrationTest -DintegrationTestPipelineOptions='[
+ * "--serverName=<YOUR SNOWFLAKE SERVER NAME>",
+ * "--username=<USERNAME>",
+ * "--privateKeyPath=<PATH TO PRIVATE KEY>",
+ * "--privateKeyPassphrase=<KEY PASSPHRASE>",
+ * "--database=<DATABASE NAME>",
+ * "--schema=<SCHEMA NAME>",
+ * "--stagingBucketName=<BUCKET NAME>",
+ * "--storageIntegrationName=<STORAGE INTEGRATION NAME>",
+ * "--snowPipe=<SNOWPIPE NAME>",
+ * "--numberOfRecords=<INTEGER>",
+ * "--runner=DirectRunner"]'
+ * --tests org.apache.beam.sdk.io.snowflake.test.SnowflakeStreamingIOIT.writeStreamThenRead
+ * -DintegrationTestRunner=direct
+ * </pre>
+ */
+public class StreamingSnowflakeIOIT {
+  private static final int TIMEOUT = 900000;
+  private static final int INTERVAL = 30000;
+  private static final String TABLE = "STREAMING_IOIT";
+
+  private static final List<TestRow> testRows = newArrayList();
+
+  @Rule public final transient TestPipeline pipeline = TestPipeline.create();
+
+  private static SnowflakeIOITPipelineOptions options;
+  private static SnowflakeIO.DataSourceConfiguration dc;
+  private static String stagingBucketName;
+  private static String storageIntegrationName;
+
+  @BeforeClass
+  public static void setupAll() throws SQLException {
+    PipelineOptionsFactory.register(SnowflakeIOITPipelineOptions.class);
+    options = TestPipeline.testingPipelineOptions().as(SnowflakeIOITPipelineOptions.class);
+
+    dc =
+        SnowflakeIO.DataSourceConfiguration.create(SnowflakeCredentialsFactory.of(options))
+            .withServerName(options.getServerName())
+            .withDatabase(options.getDatabase())
+            .withRole(options.getRole())
+            .withWarehouse(options.getWarehouse())
+            .withSchema(options.getSchema());
+
+    stagingBucketName = options.getStagingBucketName();
+    storageIntegrationName = options.getStorageIntegrationName();
+
+    for (int i = 0; i < options.getNumberOfRecords(); i++) {
+      testRows.add(TestRow.create(i, String.format("TestRow%s:%s", i, UUID.randomUUID())));
+    }
+
+    TestUtils.runConnectionWithStatement(
+        dc.buildDatasource(),
+        String.format("CREATE OR REPLACE TABLE %s(id INTEGER, name STRING)", TABLE));
+  }
+
+  @AfterClass
+  public static void cleanUp() throws Exception {
+    String bucketNameAndPath = stagingBucketName.replaceAll(".+//", "");
+    String[] parts = bucketNameAndPath.split("/", -1);
+    String bucketName = parts[0];
+    String directory = null;
```

Comment:
Maybe change to 
`String directory = parts.length > 1 ? bucketNameAndPath.replace(bucketName + "/", "") : null` ?

Reply:
I think right now it looks more clear when it's step by step.